src/test/java/seedu/address/logic/parser/ParserUtilTest.java

```diff
@@ -50,6 +55,130 @@ public void parseIndex_validInput_success() throws Exception {
+    }
+
+    @Test
+    public void parsePersonSortField_invalidValue_throwsParseException() {
+        // Invalid letter
+        assertThrows(ParseException.class, () -> ParserUtil.parsePersonSortField("x"));
+
+        // Multiple letters
+        assertThrows(ParseException.class, () -> ParserUtil.parsePersonSortField("xyz"));
+
+        // Whitespace
+        assertThrows(ParseException.class, () -> ParserUtil.parsePersonSortField(WHITESPACE));
+    }
+
+    @Test
+    public void parsePersonSortField_validValueWithoutWhitespace_returnsSortField() throws Exception {
+
+        String lowercaseName = "n";
+        String uppercaseName = "N";
+
+        String lowercaseGender = "g";
+        String uppercaseGender = "G";
+
+        String lowercaseDob = "d";
+        String uppercaseDob = "D";
+
+        // Lower case name
+        PersonSortField sortFieldLowercaseName = PersonSortField.createSortField(lowercaseName);
+        assertEquals(sortFieldLowercaseName, ParserUtil.parsePersonSortField(lowercaseName));
+
+        // Upper case name
+        PersonSortField sortFieldUppercaseName = PersonSortField.createSortField(uppercaseName);
+        assertEquals(sortFieldUppercaseName, ParserUtil.parsePersonSortField(uppercaseName));
+
+        // Lower case gender
+        PersonSortField sortFieldLowercaseGender = PersonSortField.createSortField(lowercaseGender);
+        assertEquals(sortFieldLowercaseGender, ParserUtil.parsePersonSortField(lowercaseGender));
+
+        // Upper case gender
+        PersonSortField sortFieldUppercaseGender = PersonSortField.createSortField(uppercaseGender);
+        assertEquals(sortFieldUppercaseGender, ParserUtil.parsePersonSortField(uppercaseGender));
+
+        // Lower case DOB
+        PersonSortField sortFieldLowercaseDob = PersonSortField.createSortField(lowercaseDob);
+        assertEquals(sortFieldLowercaseDob, ParserUtil.parsePersonSortField(lowercaseDob));
+
+        // Upper case DOB
+        PersonSortField sortFieldUppercaseDob = PersonSortField.createSortField(uppercaseDob);
+        assertEquals(sortFieldUppercaseDob, ParserUtil.parsePersonSortField(uppercaseDob));
+    }
+
+    @Test
+    public void parsePersonSortField_validValueWithWhitespace_returnsSortField() throws Exception {
+        String lowercaseName = "n";
+        String lowercaseNameWithWhitespace = WHITESPACE + lowercaseName + WHITESPACE;
+
+        PersonSortField sortFieldLowercaseName = PersonSortField.createSortField(lowercaseName);
+        assertEquals(sortFieldLowercaseName, ParserUtil.parsePersonSortField(lowercaseNameWithWhitespace));
+    }
+
+
+    //=========== parseEventSortField() Tests =================================================================
+
+    @Test
+    public void parseEventSortField_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> ParserUtil.parseEventSortField((String) null));
```

Comment:
I think converting null to type `String` isn't necessary since its already null
```suggestion
        assertThrows(NullPointerException.class, () -> ParserUtil.parseEventSortField(null));
```

Reply:
Yup you're right! It doesn't do anything, but I'm following how the other tests handle null values, which is to cast it to the correct argument type.